src/main/scala/Decoder.scala

```diff
@@ -0,0 +1,247 @@
+    SRLI       -> List(uOP.SRL       , InstType.calculate, ImmSel.is_shamt),
+    SRAI       -> List(uOP.SRA       , InstType.calculate, ImmSel.is_shamt),
+    ORI        -> List(uOP.OR        , InstType.calculate, ImmSel.is_I    ),
+    ANDI       -> List(uOP.AND       , InstType.calculate, ImmSel.is_I    ),
+    ADD        -> List(uOP.ADD       , InstType.calculate, ImmSel.no_imm  ),
+    SUB        -> List(uOP.SUB       , InstType.calculate, ImmSel.no_imm  ),
+    SLL        -> List(uOP.SLL       , InstType.calculate, ImmSel.no_imm  ),
+    SLT        -> List(uOP.SLT       , InstType.calculate, ImmSel.no_imm  ),
+    SLTU       -> List(uOP.SLTU      , InstType.calculate, ImmSel.no_imm  ),
+    XOR        -> List(uOP.XOR       , InstType.calculate, ImmSel.no_imm  ),
+    SRL        -> List(uOP.SRL       , InstType.calculate, ImmSel.no_imm  ),
+    SRA        -> List(uOP.SRA       , InstType.calculate, ImmSel.no_imm  ),
+    OR         -> List(uOP.OR        , InstType.calculate, ImmSel.no_imm  ),
+    AND        -> List(uOP.AND       , InstType.calculate, ImmSel.no_imm  ),
+
+    LB         -> List(uOP.LB        , InstType.load     , ImmSel.is_I    ),
+    LH         -> List(uOP.LH        , InstType.load     , ImmSel.is_I    ),
+    LW         -> List(uOP.LW        , InstType.load     , ImmSel.is_I    ),
+    LBU        -> List(uOP.LBU       , InstType.load     , ImmSel.is_I    ),
+    LHU        -> List(uOP.LHU       , InstType.load     , ImmSel.is_I    ),
+    LWU        -> List(uOP.LWU       , InstType.load     , ImmSel.is_I    ),
+    SB         -> List(uOP.SB        , InstType.store    , ImmSel.is_S    ),
+    SH         -> List(uOP.SH        , InstType.store    , ImmSel.is_S    ),
+    SW         -> List(uOP.SW        , InstType.store    , ImmSel.is_S    ),
+
+    CSRRW      -> List(uOP.CSRRW     , InstType.calculate, ImmSel.no_imm  ),
+    CSRRS      -> List(uOP.CSRRS     , InstType.calculate, ImmSel.no_imm  ),
+    CSRRC      -> List(uOP.CSRRC     , InstType.calculate, ImmSel.no_imm  ),
+    CSRRWI     -> List(uOP.CSRRW     , InstType.calculate, ImmSel.is_zimm ),
+    CSRRSI     -> List(uOP.CSRRS     , InstType.calculate, ImmSel.is_zimm ),
+    CSRRCI     -> List(uOP.CSRRC     , InstType.calculate, ImmSel.is_zimm ),
+    FENCE      -> List(uOP.FENCE     , InstType.fence    , ImmSel.no_imm  ),
+    FENCE_I    -> List(uOP.FENCE_I   , InstType.fence    , ImmSel.no_imm  ),
+    ECALL      -> List(uOP.ECALL     , InstType.system   , ImmSel.no_imm  ),
+    EBREAK     -> List(uOP.EBREAK    , InstType.system   , ImmSel.no_imm  ),
+    MRET       -> List(uOP.MRET      , InstType.system   , ImmSel.no_imm  ),
+    SRET       -> List(uOP.SRET      , InstType.system   , ImmSel.no_imm  ),
+    SFENCE_VMA -> List(uOP.SFENCE_VMA, InstType.fence    , ImmSel.no_imm  ),
+    WFI        -> List(uOP.WFI       , InstType.system   , ImmSel.no_imm  ),
+    )
+
+}
+
+object uOP extends ChiselEnum {
```

Comment:
I don't know whether `uOP` is the best name here...
However I don't have a better alternative.
Please at least make it `UOp`.